### PR TITLE
[:ensure] = :unsupported key check for beaker create_manifest

### PR DESCRIPTION
**Pull request by chrisvanheuveln**
The README changes explain it all...

### CODE REVIEW

**Comment by glennmatthews** [comment]
Is there a particular provider or test that this is being added for?

**Comment by chrisvanheuveln** [comment]
stp_global. It's a WIP.

File: docs/README-develop-beaker-scripts.md [hunk 1]
```diff
@@ -961,6 +961,7 @@ Custom keys may be added for various unique test requirements but they should be
 |:-----------------|:---------------
 | `:asn`           | An Autonomous-System number. Used in some routing protocol tests.
 | `:encap_prof_global` | The global encapsulation profile configuration. This configuration is a dependency for testing with `cisco_interface_service_vni`.
+| `:ensure`        | Allowed values: `:unsupported`. This key is only used for providers that do not support ensurability; this prevents the test_harness from inserting an ensure value into the manifest. Also see Test Case key `:ensure`.
```
Comment (glennmatthews) [inline docs/README-develop-beaker-scripts.md]: This seems a bit confusing to me (`:ensure` means one thing at top level, a different thing at test case level). Would it be clearer and more meaningful if this new top-level key is called `:ensurable`?

Comment (chrisvanheuveln) [inline docs/README-develop-beaker-scripts.md]: I had the same concerns actually. I like the :ensurable t/f suggestion.

**Comment by chrisvanheuveln** [comment]
@glennmatthews Any more comments?

File: docs/README-develop-beaker-scripts.md [hunk 2]
```diff
@@ -989,7 +990,7 @@ The keys are all optional. Most test cases will typically only use `:desc` and `
...
 | `:preclean`       | The resource name used to remove a configuration as part of testbed pre-cleanup. The resource does not have to be the same as the one being tested; for example, the `'cisco_bgp_af'` test might use `'cisco_bgp'` for `:preclean` if a full bgp cleanup is needed instead of just removing the `'cisco_bgp_af'` configuration. This key can be specified as a common top-level key or on a test-by-test basis.
 | `:resource`       | A hash of expected values from the `puppet resource` command. This is only needed when the expected values differ from the values defined by `:manifest_props`; for example: default testing may specify a value of `'default'` in the manifest but an expected value of `'true'` for the `puppet resource` command.
-| `:ensure`         | The ensure value: `'present'` or `'absent'`. If not specified the test case will default to `'present'`.
+| `:ensure`         | The ensure value: `'present'` or `'absent'`. If not specified the test case will default to `'present'`. Also see Top-level key `:ensure`.
```
Comment (glennmatthews) [inline docs/README-develop-beaker-scripts.md]: Revert this now that the top-level key is different?

**Comment by glennmatthews** [comment]
Sorry, hadn't seen the new diffs. One nit, otherwise :+1:

File: docs/README-develop-beaker-scripts.md (hunk 2 above)
Comment (chrisvanheuveln) [inline docs/README-develop-beaker-scripts.md]: They're related so I thought it would be helpful to keep it, although I should have used the new name.

**Comment by chrisvanheuveln** [comment]
Looks like somebody already picked up my :ensurable change so all that's left of this PR is the doc change.

**Pull request merged**